srcts/src/components/sidebar.ts

```diff
@@ -4,20 +4,173 @@ import {
   doWindowResizeOnElementResize,
 } from "./_utils";
 
+type SidebarMethod = "close" | "open";
+
 type MessageData = {
-  method: "close" | "open";
+  method: SidebarMethod | null;
+};
+
+type SidebarComponents = {
+  container: HTMLElement;
+  main: HTMLElement;
+  sidebar: HTMLElement;
+  toggle: HTMLElement;
+  isClosed: boolean;
 };
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-const COLLAPSE_CLASS = "sidebar-collapsed";
+class BslibSidebar {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  public static readonly COLLAPSE_CLASS = "sidebar-collapsed";
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  public static readonly LAYOUT_CLASS = "bslib-sidebar-layout";
+
+  public static initSidebar(el: HTMLElement): void {
+    const container = BslibSidebar._findLayoutContainer(el);
+    // remove script with onload attribute to signal initialization happened
+    container.removeChild(el);
```

Comment:
Maybe make this less brittle to parent-child relationship?

Reply:
Thanks to the `data-bslib-sidebar-init="{id}"` attribute, `initCollapsible()` is handed the layout container directly.

I renamed the method `initCollapsible()` because we only need to run the initialization when the sidebar is collapsible. In fact, we don't even include the initialization script if the sidebar is always visible.